dataset_api.md

```diff
@@ -0,0 +1,158 @@
+d.default_graph = g1
+
+# Add a Graph to a Named Graph in the Dataset.
+g2 = Graph()
+g2.add(
+    (
+        URIRef("http://example.com/subject-b"),
+        URIRef("http://example.com/predicate-b"),
+        Literal("Triple B")
+    )
+)
+d.add_named_graph(name=URIRef("http://example.com/graph-B"), g2)
+
+# ============================================
+# Iterate over the entire Dataset returning triples
+# ============================================
+
+for triple in d.triples():
+    print(triple)
+# Output:
+(rdflib.term.URIRef('http://example.com/subject-a'), rdflib.term.URIRef('http://example.com/predicate-a'), rdflib.term.Literal('Triple A'))
+(rdflib.term.URIRef('http://example.com/subject-b'), rdflib.term.URIRef('http://example.com/predicate-b'), rdflib.term.Literal('Triple B'))
+
+# ============================================
+# Iterate over the entire Dataset returning quads
+# ============================================
+
+for quad in d.quads():
+    print(quad)
+# Output:
+(rdflib.term.URIRef('http://example.com/subject-a'), rdflib.term.URIRef('http://example.com/predicate-a'), rdflib.term.Literal('Triple A'), None)
+(rdflib.term.URIRef('http://example.com/subject-b'), rdflib.term.URIRef('http://example.com/predicate-b'), rdflib.term.Literal('Triple B'), rdflib.term.URIRef('http://example.com/graph-B'))
+
+# ============================================
+# Get the Default graph
+# ============================================
+
+dg = d.default_graph  # same as current default_context
+
+# ============================================
+# Iterate on triples in the Default Graph only
+# ============================================
+
+for triple in d.triples(graph="default"):
+    print(triple)
+# Output:
+(rdflib.term.URIRef('http://example.com/subject-a'), rdflib.term.URIRef('http://example.com/predicate-a'), rdflib.term.Literal('Triple A'))
+
+# ============================================
+# Access quads in Named Graphs only
+# ============================================
+
+for quad in d.quads(graph="named"):
+    print(quad)
+# Output:
+(rdflib.term.URIRef('http://example.com/subject-b'), rdflib.term.URIRef('http://example.com/predicate-b'), rdflib.term.Literal('Triple B'), rdflib.term.URIRef('http://example.com/graph-B'))
+
+# ============================================
+# Equivalent to iterating over graphs()
+# ============================================
+
+for ng_name, ng_object in d.graphs().items():
+    for quad in d.quads(graph=ng_name):
+        print(quad)
+# Output:
+(rdflib.term.URIRef('http://example.com/subject-b'), rdflib.term.URIRef('http://example.com/predicate-b'), rdflib.term.Literal('Triple B'), rdflib.term.URIRef('http://example.com/graph-B'))
+
+# ============================================
+# Access triples in the Default Graph and specified Named Graphs.
+# ============================================
+
+for triple in d.triples(graph=["default", URIRef("http://example.com/graph-B")]):
```

Comment:
`d.triples()` doesn't really make sense? There should be `Graph.triples()` and `Dataset.quads()` only?

Reply:
I'm comfortable with it - SPARQL queries in triplestores where named graphs are used frequently omit the graph, only having basic graph patterns, and we understand this to be across all graphs?

Reply:
Union graph is an extension feature though, not a feature of an RDF dataset.